src/serialization/serialization.ts

```diff
@@ -36,147 +44,23 @@ export interface SlateDocument {
   object: "value";
-  data?: { [key: string]: any };
-  document?: SlateDocument;
-}
-
-function typeProp(type?: string) {
-  return type != null ? { type } : {};
-}
-
-function keyProp(key?: string) {
-  return key != null ? { key } : {};
-}
-
-export function serializeMark(mark: MarkJSON) {
-  const { type, data } = mark;
-  const dataKeys = data && keys(data);
-  const dataValues = data && values(data);
-  // special case for single data values, so we get { color: "#aabbcc" }
-  // instead of { color: { color: "#aabbcc"} }
-  return ((dataValues?.length === 1) && (dataKeys?.[0] === type))
-            ? dataValues[0]
-            : (dataValues?.length ? data : true);
+  data?: DocumentMetadata;
+  document?: BaseElement;
 }
 
-export function serializeTextNode(node: TextJSON): SlateText {
-  const { key, text, marks } = node;
-  const textNode: SlateText = { ...keyProp(key), text: text || "" };
-  marks?.forEach(mark => {
-    textNode[mark.type] = serializeMark(mark);
-  });
-  return textNode;
+export function serializeDocument(document: Descendant[]) {
+  return { children: document };
 }
 
-export function serializeChildren(nodes: NodeJSON[], objTypes: ObjectTypeMap): SlateNode[] {
-  return nodes.map(node => serializeNode(node, objTypes));
-}
-
-// The 0.47 notions of Blocks/Inlines are combined in 0.50 to the notion of Element
-export function serializeElement(node: ElementJSON, objTypes: ObjectTypeMap): SlateElement {
-  const { object, type, key, nodes, data } = node;
-  const children = serializeChildren(castArray(nodes), objTypes);
-  const element: SlateElement = { ...typeProp(type), ...keyProp(key), children, ...data };
-  object && (objTypes[type] = object);
-  return element;
-}
-
-export function serializeNode(node: NodeJSON, objTypes: ObjectTypeMap): SlateNode {
-  const { object } = node;
-  switch (object) {
-    case "block":
-    case "inline":
-      return serializeElement(node as ElementJSON, objTypes);
-    case "text":
-      return serializeTextNode(node as TextJSON);
-    default:
-      return (node as ElementJSON)?.nodes
-              ? serializeElement(node as ElementJSON, objTypes)
-              : serializeTextNode(node as TextJSON);
-  }
-}
-
-export function serializeDocument(document: DocumentJSON): SlateDocument {
-  const { nodes, key, data } = document;
-  const objTypes: ObjectTypeMap = {};
-  const children = serializeChildren(castArray(nodes), objTypes);
-  // return objTypes map as part of document for use in deserialization
-  return { ...keyProp(key), children, objTypes, ...data };
-}
-
-export function serializeSelection(value: Value) {
-  const { document, selection } = value;
-  const nodes = document.getDescendantsAtRange(selection).toArray();
-  const objTypes: ObjectTypeMap = {};
-  return nodes.map(node => serializeNode(node.toJSON(), objTypes));
-}
+// FIXME: This needs to be updated to the new slate API.
+// export function serializeSelection(value: Value) {
```

Comment:
I think this can be updated to take the Editor.selection (a Range) and to wrap that up in a way similar to  `serializeDocument` does